test_framework/tacky/common.py

```diff
@@ -220,3 +220,17 @@ def is_zero_instr(i: asm.AsmItem) -> bool:
         and i.opcode == Opcode.XOR
         and i.operands[0] == i.operands[1]
     )
+
+def check_constant_folds(parsed_asm, program_source_file):
+    for fn_name, fn_body in parsed_asm.items():
+        if fn_name.startswith("target"):
+            bad_instructions = [i for i in fn_body.instructions if not ok(i)]
```

Comment:
`ok` isn't defined here - the two callers define slightly different `ok` functions. We'd need to pass `ok` as an argument here